frontend/src/components/managePermissions/DataTableHeader.vue

```diff
@@ -0,0 +1,153 @@
+<script setup lang="ts">
+import InputText from 'primevue/inputtext';
+import router from '@/router';
+import { IconSize } from '@/enum/IconEnum';
+import { selectedApplicationDisplayText , filters } from '@/store/ApplicationState';
+
+const props = defineProps({
+    btnLabel: {
+        type: String,
+        required: true,
+    },
+    btnRoute: {
+        type: String,
+        required: true
+    }
+});
+
+</script>
+
+<template>
+    <div class="custom-data-table-header">
+        <h3>{{ selectedApplicationDisplayText }} users</h3>
+        <span>
+            This table shows all the users in
+            {{ selectedApplicationDisplayText }} and their permissions
+            levels
+        </span>
+    </div>
+
+    <div class="search-container">
+        <Button
+            class="btn-add-user"
+            :label="props.btnLabel"
+            @click="router.push(props.btnRoute)"
+        >
+            <Icon icon="add" :size="IconSize.small" />
+        </Button>
+        <span class="p-input-icon-left">
+            <Icon icon="search" :size="IconSize.small" />
+            <InputText
+                id="dashboardSearch"
+                class="dash-search"
+                placeholder="Search by keyword"
+                v-model="filters['global'].value"
```

Comment:
the search is not working anymore

Reply:
I'll take a look on what is going on

Reply:
I wonder if this is caused by the filters... you have a global filter in the store, but also have a separate admin filter in the ApplicationAdminTable. Maybe it's better the pass the filter as a property.

Reply:
I already found the problem, I'm just testing to see if I can reuse without problem